### PR TITLE
Allow associations to be iterated

### PR DESCRIPTION
**Pull request by davidyell**
Added an iterator to the class so that associations can be looped through

### CODE REVIEW

File: src/ORM/AssociationCollection.php [hunk 1]
```diff
@@ -292,4 +294,14 @@ public function normalizeKeys($keys)
 
         return $this->_normalizeAssociations($keys);
     }
+    
+    /**
+     * Allow looping through the associations
+     * 
+     * @return \Cake\ORM\Association
```
Comment (lorenzo) [inline src/ORM/AssociationCollection.php]: It returns an ArrayIterator

**Comment by lorenzo** [comment]
A test would be nice :)

File: src/ORM/AssociationCollection.php (hunk 1 above)
Comment (davidyell) [inline src/ORM/AssociationCollection.php]: Oh yeah, lol, the ArrayIterator will return the associations.

File: tests/TestCase/ORM/AssociationCollectionTest.php [hunk 2]
```diff
@@ -367,4 +367,25 @@ public function testNormalizeKeys()
...
+        $belongsToMany = new BelongsToMany('');
+        $this->associations->add('Cart', $belongsToMany);
+
+        foreach ($this->associations as $name => $association) {
+            if ($name === 'users') {
+                $this->assertInstanceOf('Cake\ORM\Association\BelongsTo', $association);
+            } else {
+                $this->assertInstanceOf('Cake\ORM\Association\BelongsToMany', $association);
+            }
+        }
```
Comment (lorenzo) [inline tests/TestCase/ORM/AssociationCollectionTest.php]: A way to simplify this is:

```php
$expected = ['users' => $belongsTo, 'cart' => $belongsToMany];
$result = iterator_to_array($this->associations, true);
$this->assertSame($expected, $result);
```

Comment (davidyell) [inline tests/TestCase/ORM/AssociationCollectionTest.php]: This doesn't feel like it's actually doing any iteration at all though, as it's just an array comparison.

File: src/ORM/AssociationCollection.php [hunk 3]
```diff
@@ -292,4 +294,14 @@ public function normalizeKeys($keys)
 
         return $this->_normalizeAssociations($keys);
     }
+
+    /**
+     * Allow looping through the associations
+     *
+     * @return ArrayIterator
```
Comment (ADmad) [inline src/ORM/AssociationCollection.php]: `@return \ArrayIterator`

**Comment by lorenzo** [comment]
Thanks @davidyell

**Pull request merged**